src/common/ad_mem.c

```diff
@@ -1618,6 +1822,10 @@ arena_reserve_addr(struct ad_arena *arena, daos_size_t size, struct ad_reserv_ac
 	}
 	/* NB: reorder only does minimum amount of works most of the time */
 	arena_reorder_grp(arena, grp, grp_at, true);
+	node = arena2heap_node(arena);
+	D_ASSERT(node->mh_free_size >= grp->gp_df->gd_unit);
+	node->mh_free_size -= grp->gp_df->gd_unit;
+	arena_reorder_if_needed(arena);
```

Comment:
Sure, in arena_reoder_if_needed(), it will return directly without updating position since node->mh_active is true. and mh_in_tree is 0.

Let me remove this call which will make it more clear we don't update.